### ISSUE TITLE
List of issues and changes

### ISSUE THREAD

**Issue by ghost** [revolucas/CoC-Xray#16]
Posting here if anyone wants to fix something.
- ~~actor bleeding can be overwritten with smaller one, should keep highest or add up with a cap~~ https://github.com/revolucas/CoC-Xray/commit/f0a2cadddbe457ff4af4d5df87a4a5821d6af2f4
- ~~items with uses don't update their weight on load, only after opening inventory~~ (https://github.com/revolucas/CoC-Xray/commit/17e45b96dc1d3049a4050ddd52f49437039a4367)
- ~~weapons with tri state reload can get stuck in jammed if the magazine is full (have to unload), jamming  should happen after shot IMO; jam fix should require full reload (for OWR3)~~ (FIXED: https://github.com/revolucas/CoC-Xray/commit/02400c2bbbd28038d02df0aaed48233da54b279a)
- not sure if still exists; recoil [bug 1](http://www.coc.epicstalker.com/bugtracker/view.php?id=810); [bug 2](http://www.coc.epicstalker.com/bugtracker/view.php?id=811)
- ~~burn damage slows the actor~~
- live detector selects targets ahead and behind the camera
- bullet pass through is not enabled (3rd param in damages.ltx)
- visible shadows disappear when using fov higher than 55
- distant shadow bug caused by cloud shadows: https://youtu.be/H2YB0yxFFtM & https://youtu.be/Q_O5D-S0GVk; speed of these shadows is also too high, especially in stormy/windy weather
- pause can temporarily break sounds, near anomalies for example: https://youtu.be/nbw8cCYVda4

other changes or stuff that should be added:
- (Alun's idea) actor legs
- kick sound; needs at least two, hit and miss
- handling and recoil modifier for weapon attachments (simulate adding weight); more important due to weapon intertion
OR
- (Alun's idea) weight automatically changes recoil and handling (or anything else that depends on it); scale is not 1:1, should be adjustable through config
- port g_hardcore from [Oxygen](https://github.com/xrOxygen/xray-oxygen)
- port 3D and secondary scope from Shoker
- restore cut items: dosimeter (port from [Oxygen](https://github.com/xrOxygen/xray-oxygen/commits?author=Unfainthful)?), anomaly detector, live detector (NPC counter); would help equipment progression by not having these default on and even having them tiered like artefact detectors or binoculars
- staggering kick ability for actor (no or minimal damage); NPCs punch, you kick them in the nut!
- ~~upgrade support for binoculars (starts with fixed zoom and no live detector)~~
- ~~toggle for night vision scopes (including binoculars)~~
- ~~minimum zoom value for weapon/equipment with dynamic zoom~~ https://github.com/revolucas/CoC-Xray/commit/38731ae322ea9c6712452d012d36b7c73700f37d
- ~~movement speed modifier when aiming while crouching~~ https://github.com/revolucas/CoC-Xray/commit/b170c4b86bdcff4f59a11dacb804becfcdbd577a
- ~~k_cam_dispersion recoil modifier for ammo~~ https://github.com/revolucas/CoC-Xray/commit/634628471a4af4b6831bffbdd60f9d4856652af5
- ~~AP modifier for weapons (lower bullet speed -> less penetration)~~ https://github.com/revolucas/CoC-Xray/commit/3bc62332ac452bd22a7d1d47cb2d18a5a29b4042
- possibility to disable actor dispersion for ammo (buckshot no longer changes spread) or PDM multis for ammo
- broken state for weapons when they can't be used at all (maybe if jams under misfire_end_condition?); weapon explodes instead of jamming if condition is 0%
- bullet sound configurable per ammo
- range slider and fade for grass shadows
- ~~(idea) HUD customisation: enable or disable part(s) of the HUD, including NPC counter sound~~
- ~~(idea) grab and move all dynamic objects (with weight limit ofc)~~ (Alun: This can be done in system.ltx and object\models config with correct capture settings)
- (idea) support for multiple cloud shadow textures that changes with weather
Full list of my brain farts: https://pastebin.com/FX5g41PA

**Comment by ghost** [comment]
Can it be compiled under Linux?

I managed to build OpenX-Ray, but AFAIK that only works with CoP, and I want to play CoC :)

**Comment by revolucas** [comment]
No. But it runs great in WINE. Here is tutorial to set it up: https://www.moddb.com/tutorials/running-call-of-chernobyl-on-linux-with-wine

**Comment by ghost** [comment]
Thanks, I know it runs great in WINE, I've achieved legendary status in CoC that way :)

If CoP can run natively, then surely CoC can too, since they share the same engine, and I have a hard time imagining that any of the CoC-specific code is bound to Windows?...

**Comment by revolucas** [comment]
CoP can only run natively in Windows. The original engine only supports DirectX and Direct3D. Open X-ray is a heavily modified version of the source code with dozens of people working on it. It's OpenGL support, last I remember, wasn't fully implemented but it does compile and run on Linux. CoC is a heavily modified version of the original source code and CoC requires these changes and it would take a lot of work to port these changes to Open X-ray. It's not something I'm interested in.

**Comment by Shtrecker** [comment]
Можете помочь с настройкой Visual Studio для правильной сборки? Пожалуйста.
Can you help with configuring Visual Studio for the correct build? Please.

**Comment by Shtrecker** [comment]
![visual studio](https://user-images.githubusercontent.com/77389805/131217407-6f766507-70f1-4acd-ae58-42850c86240c.png)
